### PR TITLE
PAL-506-Readers-remove-unused-dependencies: pom files with dependenci…

### PR DESCRIPTION
**Pull request by m78233**
…es removed.

Signed-off-by: m78233 <61839550+m78233@users.noreply.github.com>

### CODE REVIEW

File: hadoop-reader/pom.xml [hunk 1]
```diff
@@ -26,47 +26,37 @@
...
-        <fabric8.version>4.3.0</fabric8.version>
-        <protobuf.version>3.5.1</protobuf.version>
         <junit.version>4.13.1</junit.version>
-        <mockito.version>3.1.0</mockito.version>
-        <hamcrest.version>2.1</hamcrest.version>
         <logback.version>1.2.3</logback.version>
     </properties>
 
     <dependencies>
+        <!-- User, Resource, Context types -->
```
Comment (pd104923) [inline hadoop-reader/pom.xml:41]: I think this needs to be re-worded as `readers-common` doesn't contain these "things". Readers-common contains the readers, writers and the requests objects.

**Review by pd104923 — CHANGES_REQUESTED** [review]
One minor tweak

File: hadoop-reader/pom.xml (hunk 1 above)
Comment (m78233) [reply hadoop-reader/pom.xml]: Yes,  changed to Readers, Request and Response

**Review by pd104923 — APPROVED** [review]

File: hadoop-reader/pom.xml [hunk 2]
```diff
@@ -78,24 +68,13 @@
                 </exclusion>
             </exclusions>
         </dependency>
+        <!-- JUnit 4 test framework -->
```
Comment (developer01189998819991197253) [inline hadoop-reader/pom.xml:71]: Can just replace with
```suggestion
        <!-- Testing Dependency -->
```
Saves us having to update it when we update it to junit 5

File: hadoop-resource/pom.xml [hunk 3]
```diff
@@ -59,45 +59,20 @@
                 </exclusion>
             </exclusions>
         </dependency>
-        <dependency>
-            <groupId>uk.gov.gchq.palisade</groupId>
-            <artifactId>common</artifactId>
-            <version>0.5.0-${common.revision}</version>
-        </dependency>
+        <!-- JUnit 5 test framework -->
```
Comment (developer01189998819991197253) [inline hadoop-resource/pom.xml:62]: same here, just change it to "Testing Dependency"

**Review by developer01189998819991197253 — CHANGES_REQUESTED** [review]

File: hadoop-resource/pom.xml (hunk 3 above)
Comment (m78233) [reply hadoop-resource/pom.xml:62]: done

**Review by developer01189998819991197253 — APPROVED** [review]

**Review by pd104923 — DISMISSED** [review]

**Review by dev930018 — COMMENTED** [review]
I'd like to see a build of services with these changes - I remember hadoop being quite finicky

File: hadoop-reader/pom.xml [hunk 4]
```diff
@@ -26,47 +26,37 @@
...
-        <fabric8.version>4.3.0</fabric8.version>
-        <protobuf.version>3.5.1</protobuf.version>
         <junit.version>4.13.1</junit.version>
-        <mockito.version>3.1.0</mockito.version>
-        <hamcrest.version>2.1</hamcrest.version>
         <logback.version>1.2.3</logback.version>
     </properties>
 
     <dependencies>
+        <!-- Readers, Request & Response types -->
```
Comment (dev930018) [inline hadoop-reader/pom.xml:41]: Could be styled more like services are, but not too big an issue

File: hadoop-reader/pom.xml [hunk 5]
```diff
@@ -26,47 +26,37 @@
...
             <groupId>ch.qos.logback</groupId>
             <artifactId>logback-classic</artifactId>
             <version>${logback.version}</version>
         </dependency>
+        <!-- Jackson Dependency -->
         <dependency>
             <groupId>com.fasterxml.jackson.core</groupId>
             <artifactId>jackson-annotations</artifactId>
             <version>${jackson.version}</version>
         </dependency>
```
Comment (dev930018) [inline hadoop-reader/pom.xml:58]: We can remove this (and the jackson annotations) as we no longer have a need to serialise the reader object

File: hadoop-reader/pom.xml [hunk 6]
```diff
@@ -78,24 +68,13 @@
                 </exclusion>
```
Comment (dev930018) [inline hadoop-reader/pom.xml:68]: Hadoop resource service also excludes slf4j-log4j - could we add that for consistency? (It is sometimes needed to be excluded)

File: hadoop-reader/pom.xml [hunk 7]
```diff
@@ -78,24 +68,13 @@
...
             </exclusions>
         </dependency>
+        <!-- Testing Dependency -->
         <dependency>
             <groupId>junit</groupId>
             <artifactId>junit</artifactId>
             <version>${junit.version}</version>
             <scope>test</scope>
         </dependency>
```
Comment (dev930018) [inline hadoop-reader/pom.xml:77]: Is there a ticket for junit5 stuff?

File: hadoop-reader/pom.xml [hunk 8]
```diff
@@ -26,47 +26,37 @@
...
-            <groupId>uk.gov.gchq.palisade</groupId>
-            <artifactId>readers-common</artifactId>
-            <version>${project.parent.version}</version>
-        </dependency>
-        <dependency>
-            <groupId>uk.gov.gchq.palisade</groupId>
-            <artifactId>shade-hadoop-client</artifactId>
-            <version>${project.parent.version}</version>
-        </dependency>
+        <!-- Hadoop framework -->
```
Comment (dev930018) [inline hadoop-reader/pom.xml:59]: These comments could be better

File: hadoop-resource/pom.xml [hunk 9]
```diff
@@ -28,22 +28,22 @@
...
         <hadoop.version>3.2.1</hadoop.version>
-        <fabric8.version>4.3.0</fabric8.version>
-        <protobuf.version>3.5.1</protobuf.version>
-        <jetcd.version>0.3.0</jetcd.version>
         <junit.version>5.5.2</junit.version>
         <assertj.version>3.18.0</assertj.version>
-        <mockito.version>3.1.0</mockito.version>
-        <hamcrest.version>2.1</hamcrest.version>
-        <logback.version>1.2.3</logback.version>
     </properties>
```
Comment (dev930018) [inline hadoop-resource/pom.xml:33]: properties should be after artifactId and url

File: hadoop-resource/pom.xml [hunk 10]
```diff
@@ -59,45 +59,20 @@
                 </exclusion>
             </exclusions>
         </dependency>
-        <dependency>
-            <groupId>uk.gov.gchq.palisade</groupId>
-            <artifactId>common</artifactId>
-            <version>0.5.0-${common.revision}</version>
-        </dependency>
+        <!-- Testing Dependency -->
```
Comment (dev930018) [inline hadoop-resource/pom.xml:62]: Again, descriptions and style could be improved

File: pom.xml [hunk 11]
```diff
@@ -42,30 +42,29 @@
     </licenses>
 
     <properties>
-        <!-- Versions -->
+        <!-- Dependency properties -->
         <java.version>11</java.version>
         <revision>SNAPSHOT</revision>
         <common.revision>SNAPSHOT</common.revision>
+        <jackson.version>2.10.0</jackson.version>
+
```
Comment (dev930018) [inline pom.xml:50]: This won't be needed

File: pom.xml [hunk 12]
```diff
@@ -42,30 +42,29 @@
     </licenses>
 
     <properties>
-        <!-- Versions -->
+        <!-- Dependency properties -->
```
Comment (dev930018) [inline pom.xml:45]: These are better described as Properties, but nothing to do with dependencies

File: pom.xml [hunk 13]
```diff
@@ -42,30 +42,29 @@
...
-        <depends.maven.version>3.1.1</depends.maven.version>
-
-        <!-- Dependencies -->
-        <jackson.version>2.10.0</jackson.version>
+        <depends.maven.version>3.1.2</depends.maven.version>
 
-        <!-- Testing properties -->
+        <!-- Testing versions -->
         <surefire.plugin.version>2.22.1</surefire.plugin.version>
         <failsafe.version>2.22.1</failsafe.version>
```
Comment (dev930018) [inline pom.xml:65]: The failsafe version and plugin are redundant here

File: pom.xml [hunk 14]
```diff
@@ -42,30 +42,29 @@
     </licenses>
 
     <properties>
```
Comment (dev930018) [inline pom.xml:44]: I can't comment directly, but it looks like the shade-hadoop-client is now redundant and can be deleted

File: shade-hadoop-client/pom.xml [hunk 15]
```diff
@@ -36,7 +36,6 @@
     <!-- Activating the aws profile ensures the Amazon Web Services s3a connector is available in Palisade. -->
 
     <properties>
```
Comment (dev930018) [inline shade-hadoop-client/pom.xml:38]: This whole module can be deleted

File: readers-common/pom.xml [hunk 16]
```diff
@@ -25,36 +25,22 @@
...
-            <scope>test</scope>
-        </dependency>
-        <dependency>
-            <groupId>org.assertj</groupId>
-            <artifactId>assertj-core</artifactId>
-            <version>${assertj.version}</version>
-            <scope>test</scope>
+            <groupId>com.fasterxml.jackson.core</groupId>
+            <artifactId>jackson-annotations</artifactId>
+            <version>${jackson.version}</version>
```
Comment (dev930018) [inline readers-common/pom.xml:43]: Not needed, just drop the jackson annotations, we don't need to serialise this stuff

**Review by dev930018 — CHANGES_REQUESTED** [review]
Some preliminary local changes suggest something like:
```
Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
	deleted:    licenses/bsd_license_3.html
	deleted:    licenses/mit_license.html
	deleted:    shade-hadoop-client/README.md
	deleted:    shade-hadoop-client/mvn_dependency_tree.txt
	deleted:    shade-hadoop-client/pom.xml

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   NOTICES.md
	modified:   hadoop-reader/mvn_dependency_tree.txt
	modified:   hadoop-reader/pom.xml
	modified:   hadoop-reader/src/main/java/uk/gov/gchq/palisade/reader/HadoopDataReader.java
	modified:   hadoop-resource/mvn_dependency_tree.txt
	modified:   hadoop-resource/pom.xml
	modified:   pom.xml
	modified:   readers-common/mvn_dependency_tree.txt
	modified:   readers-common/pom.xml
	modified:   readers-common/src/main/java/uk/gov/gchq/palisade/reader/common/SerialisedDataReader.java
	modified:   readers-common/src/main/java/uk/gov/gchq/palisade/reader/common/SerialisingResponseWriter.java

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	licenses/eclipse public license v2.0 - epl-v20.html
```
So some license updates to do also

File: hadoop-reader/pom.xml (hunk 4 above)
Comment (m78233) [reply hadoop-reader/pom.xml:41]: changed style to be in-line with services

File: hadoop-reader/pom.xml (hunk 8 above)
Comment (m78233) [reply hadoop-reader/pom.xml:59]: Reads "Hadoop config classes" now

File: hadoop-reader/pom.xml (hunk 6 above)
Comment (m78233) [reply hadoop-reader/pom.xml:68]: added the exclusion for slf4j-log4j12  but also note i have left the original exclusion for log4j as I assume we still want to exclude this as well

File: hadoop-reader/pom.xml (hunk 7 above)
Comment (m78233) [reply hadoop-reader/pom.xml:77]: Yes, PAL-767 is in the backlog

File: hadoop-resource/pom.xml (hunk 9 above)
Comment (m78233) [reply hadoop-resource/pom.xml:33]: done

File: hadoop-resource/pom.xml (hunk 10 above)
Comment (m78233) [reply hadoop-resource/pom.xml:62]: done

File: pom.xml (hunk 14 above)
Comment (m78233) [reply pom.xml:44]: removed it

File: pom.xml (hunk 12 above)
Comment (m78233) [reply pom.xml:45]: Changed convention to the same used in services

File: pom.xml (hunk 11 above)
Comment (m78233) [reply pom.xml:50]: done

File: pom.xml (hunk 13 above)
Comment (m78233) [reply pom.xml:65]: removed both

File: readers-common/pom.xml (hunk 16 above)
Comment (m78233) [reply readers-common/pom.xml:43]: done

File: shade-hadoop-client/pom.xml (hunk 15 above)
Comment (m78233) [reply shade-hadoop-client/pom.xml:38]: done

File: pom.xml [hunk 17]
```diff
@@ -181,11 +175,6 @@
                     </execution>
                 </executions>
             </plugin>
-            <plugin>
-                <groupId>org.apache.maven.plugins</groupId>
-                <artifactId>maven-surefire-plugin</artifactId>
-                <version>${surefire.plugin.version}</version>
-            </plugin>
```
Comment (dev930018) [inline pom.xml:188]: I believe we should be keeping surefire as it's still relevant (we're still doing unit-tests)

File: hadoop-reader/pom.xml (hunk 5 above)
Comment (dev930018) [reply hadoop-reader/pom.xml]: Nope, because the annotations are used, but unneccesarily.

File: readers-common/pom.xml [hunk 18]
```diff
@@ -25,36 +25,16 @@
...
     </parent>
 
-    <properties>
-        <junit.version>5.7.1</junit.version>
-        <assertj.version>3.19.0</assertj.version>
-    </properties>
-    <!--  *********** Aritfact ID ***********    -->
+    <!--  *********** Artifact ID ***********    -->
     <artifactId>readers-common</artifactId>
     <url>https://github.com/gchq/Palisade-readers/tree/develop/readers-common</url>
```
Comment (dev930018) [inline readers-common/pom.xml:30]: Once surefire-plugin has been re-added, I've noticed readers-common has `src/test/uk...` - this should be moved to `src/test/java/uk...`

**Review by dev930018 — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 17 above)
Comment (m78233) [reply pom.xml]: yes, put it back in

File: readers-common/pom.xml (hunk 18 above)
Comment (m78233) [reply readers-common/pom.xml:30]: refactored to have java directory at the base

**Review by dev930018 — CHANGES_REQUESTED** [review]
One final thing - there's still a couple of unneccessary (and now not-depended-upon) jackson-annotations bits lying around in HadoopResourceService, HadoopDataReader and SerialisedDataReader. These should be deleted.

File: pom.xml [hunk 19]
```diff
@@ -337,7 +334,7 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-failsafe-plugin</artifactId>
-                <version>${failsafe.version}</version>
+                <version>${surefire.plugin.version}</version>
             </plugin>
```
Comment (dev930018) [inline pom.xml:338]: Failsafe plugin should be deleted if we have no component/contract tests

**Review by dev930018 — CHANGES_REQUESTED** [review]

**Review by dev930018 — APPROVED** [review]

**Pull request merged**